doc/pbh/pbh-design.md

```diff
@@ -0,0 +1,770 @@
+    - [3.1 Unit tests](#31-unit-tests)
+    - [3.2 Data plane tests](#32-data-plane-tests)
+
+## Revision
+
+| Rev | Date       | Author         | Description     |
+|:---:|:----------:|:--------------:|:----------------|
+| 0.1 | 15/03/2021 | Nazarii Hnydyn | Initial version |
+
+## About this manual
+
+This document provides general information about PBH implementation in SONiC
+
+## Scope
+
+This document describes the high level design of PBH feature in SONiC
+
+**In scope:**  
+1. PBH for NVGRE/VxLAN packets based on inner 5-tuple (IP proto, L4 dst/src port, IPv4/IPv6 dst/src)
+
+**Out of scope:**  
+1. CRM support for PBH FG hash resources
+
+## Abbreviations
+
+| Term   | Meaning                                                    |
+|:-------|:-----------------------------------------------------------|
+| SONiC  | Software for Open Networking in the Cloud                  |
+| PBH    | Policy Based Hashing                                       |
+| CRM    | Critical Resource Monitoring                               |
+| ACL    | Access Control List                                        |
+| SAI    | Switch Abstraction Interface                               |
+| FG     | Fine-Grained                                               |
+| API    | Application Programming Interface                          |
+| CRC    | Cyclic Redundancy Check                                    |
+| ID     | Identifier                                                 |
+| ECMP   | Equal-Cost Multi-Path                                      |
+| LAG    | Link Aggregation Group                                     |
+| NVGRE  | Network Virtualization Using Generic Routing Encapsulation |
+| VxLAN  | Virtual eXtensible Local Area Network                      |
+| GRE    | Generic Routing Encapsulation                              |
+| OA     | Orchestration agent                                        |
+| DB     | Database                                                   |
+| CLI    | Сommand-line Interface                                     |
+| DPB    | Dynamic Port Breakout                                      |
+| YANG   | Yet Another Next Generation                                |
+
+## List of figures
+
+[Figure 1: PBH design](#figure-1-pbh-design)  
+[Figure 2: PBH OA design](#figure-2-pbh-oa-design)  
+[Figure 3: PBH add flow](#figure-3-pbh-add-flow)  
+[Figure 4: PBH remove flow](#figure-4-pbh-remove-flow)
+
+## List of tables
+
+[Table 1: Event logging](#table-1-event-logging)
+
+# 1 Introduction
+
+## 1.1 Feature overview
+
+PBH is a feature which allows user to configure a custom hashing for different packet types.  
+Under the hood is uses ACL rules to match the specific types of frames and calculates hash  
+based on user-defined rules.
+
+For flexible hash calculation a new SAI FG Hash API is used.  
+It allows user not only to configure which fields should be used,  
+but also to specify a mask for IPv4/IPv6 addresses and sequence ID.  
```

Comment:
what is the significance of sequence Id, list out few use cases where does it use it?

Reply:
@madhupalu please have a look at:
https://github.com/Azure/SONiC/blob/6af2959ee5829f801409fc833389e78802d5258a/doc/pbh/pbh-design.md#242-configuration-sample

```json
"inner_l4_dst_port": {
    "hash_field": "INNER_L4_DST_PORT",
    "sequence_id": "2"
},
"inner_l4_src_port": {
    "hash_field": "INNER_L4_SRC_PORT",
    "sequence_id": "2"
},
```

So here we have dst/src L4 port configured as associative fields.
Which means that if values are swapped, the resulting hash will be the same.

Reply:
sounds good, thanks